core/src/main/java/org/apache/brooklyn/core/config/internal/AbstractConfigMapImpl.java

```diff
@@ -107,4 +264,264 @@ public int size() {
+                    if (!raw) result = resolveRawValueFromContainer(input, queryKey, result);
+                    return result;
+                }
+            };
+            Function<TContainer, TContainer> parentFn = new Function<TContainer, TContainer>() {
+                @Override public TContainer apply(TContainer input) {
+                    return getParentOfContainer(input);
+                }
+            };
+            AncestorContainerAndKeyValueIterator<TContainer, T> ckvi = new AncestorContainerAndKeyValueIterator<TContainer,T>(
+                getContainer(), keyFn, lookupFn, coerceFn, parentFn);
+            
+            return ConfigInheritances.resolveInheriting(
+                getContainer(), ownKey, coerceFn.apply(lookupFn.apply(getContainer())), defaultValue, 
+                ckvi, InheritanceContext.RUNTIME_MANAGEMENT, getDefaultRuntimeInheritance());
+            
+        } else {
+            String message = "Config key "+ownKey+" of "+getBrooklynObject()+" is not a ConfigKeySelfExtracting; cannot retrieve value; returning default";
+            LOG.warn(message);
+            return ReferenceWithError.newInstanceThrowingError(new BasicConfigValueAtContainer<TContainer,T>(getContainer(), ownKey, null, false,
+                    defaultValue),
+                new IllegalStateException(message));
+        }
+    }
+    
+    @Override
+    public List<ConfigValueAtContainer<TContainer,?>> getConfigAllInheritedRaw(ConfigKey<?> queryKey) {
+        List<ConfigValueAtContainer<TContainer, ?>> result = MutableList.of();
+        TContainer c = getContainer();
+        int count=0;
+
+        final InheritanceContext context = InheritanceContext.RUNTIME_MANAGEMENT;
+        ConfigInheritance currentInheritance = ConfigInheritances.findInheritance(queryKey, context, getDefaultRuntimeInheritance());
+        
+        BasicConfigValueAtContainer<TContainer, Object> last = null;
+        
+        while (c!=null) {
+            Maybe<Object> v = getRawValueAtContainer(c, queryKey);
+            BasicConfigValueAtContainer<TContainer, Object> next = new BasicConfigValueAtContainer<TContainer, Object>(c, getKeyAtContainer(c, queryKey), v);
+            
+            if (last!=null && !currentInheritance.considerParent(last, next, context)) break;
+            
+            ConfigInheritance currentInheritanceExplicit = ConfigInheritances.findInheritance(next.getKey(), InheritanceContext.RUNTIME_MANAGEMENT, null);
+            if (currentInheritanceExplicit!=null) {
+                if (count>0 && !currentInheritanceExplicit.isReinheritable(next, context)) break;
+                currentInheritance = currentInheritanceExplicit;
+            }
+            
+            if (next.isValueExplicitlySet()) result.add(0, next);
+
+            last = next;
+            c = getParentOfContainer(c);
+            count++;
+        }
+        
+        return result;
+    }
```

Comment:
There feels like a scary looking number of methods on here now. I'm not suggesting any specific method(s) that should be removed, but it's something that we should be aware of - gut feel is that we have too many methods (even ignoring the deprecated ones).

If we really need all these different ways of access it, then another would would be to present this as "views" - e.g. a raw view, a resolved view, a local-only view, etc.

But any such change are much longer term, rather than part of this PR!